### PR TITLE
[homework #8] - js advanced-1 by Brisa Rojas

### PR DESCRIPTION
**Pull request by bytheseaside**
Hi Ilya, I'm having trouble with the tasks involving dates but thought to push and PR the other ones I have completed.

About the 3rd and 4th task, what's the **moment function** supposed to do exactly? I'm not quite sure I get it

### CODE REVIEW

**Comment by ilya-ischenko** [comment]
> Hi Ilya, I'm having trouble with the tasks involving dates but thought to push and PR the other ones I have completed.
> 
> About the 3rd and 4th task, what's the **moment function** supposed to do exactly? I'm not quite sure I get it

In case with 3rd and 4th tasks moment is JS library for working with dates, but it does not matter in this task, you can avoid using it.

**Comment by bytheseaside** [comment]
@ilya-ischenko  Hi Ilya, I finished the remaining tasks, have any feedback to give me?

File: homeworks/brisa-rojas/08-js-advanced-1/task3.js [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+let person = new NamedOne('John', 'Doe');
+console.log(person.fullName); // John Doe
+person.firstName = '';
+console.log(person.fullName); // Doe
+person.firstName = 'I used';
+person.lastName = 'a Multi Word String';
+console.log(person.fullName); // I used a Multi Word String
+person.fullName = 'Disaster happens when given a 3 words or longer name for full name';
+console.log(person.fullName); // Disaster happens [when given a 3 words or longer name for full name is missing]
+
```
Comment (ilya-ischenko) [inline homeworks/brisa-rojas/08-js-advanced-1/task3.js:159]: I think it is ok, as well as in most of cases first name or last name consists of one word and another cases was not required in this task

File: package.json [hunk 2]
```diff
@@ -17,6 +17,7 @@
   },
   "homepage": "https://github.com/qaprosoft/react-laba-arg-1#readme",
   "dependencies": {
+    "moment": "^2.29.4",
     "prettier": "^2.7.1"
   }
 }
```
Comment (ilya-ischenko) [inline package.json:23]: You added "moment" dependency to common package, remove it, please.
It you want to add some dependencies, init it in your HW folder, please.

**Review by ilya-ischenko — COMMENTED** [review]
It looks very good as for me!
Only make sure that your code is formatted properly, as for me it is not.

File: homeworks/brisa-rojas/08-js-advanced-1/task3.js [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+      if (typeof obj[keys[0]] === 'object') {
+        searchedValue = pluck(obj[keys[0]], keys.splice(1).join('.'));
+      } else {
+        searchedValue = obj[keys[0]];
+      }
+    }
+  }
+
+  return searchedValue;
+}
```
[the remark this answers is not in the document]
Comment (bytheseaside) [reply homeworks/brisa-rojas/08-js-advanced-1/task3.js:32]: Because typeof null === 'object' is true

File: package.json (hunk 2 above)
Comment (bytheseaside) [reply package.json:23]: Will do!

**Comment by bytheseaside** [comment]
>  Only make sure that your code is formatted properly, as for me it is not.

I pushed it again just to be sure, but in my IDE it was always formatted with prettier!

File: homeworks/brisa-rojas/08-js-advanced-1/task3.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ilya-ischenko) [reply homeworks/brisa-rojas/08-js-advanced-1/task3.js:32]: Yes, good

**Comment by bytheseaside** [comment]
@ilya-ischenko  I thinks it's all ok now, should I correct anything else?

**Pull request merged**